sdks/java/core/src/test/java/org/apache/beam/sdk/io/WriteTest.java

```diff
@@ -203,13 +269,20 @@ private static void runWrite(
     }
 
     TestSink sink = new TestSink();
+    Write.Bound<String> write = Write.to(sink);
+    if (numConfiguredShards != null) {
+      write = write.withNumShards(numConfiguredShards);
+    }
     p.apply(Create.timestamped(inputs, timestamps).withCoder(StringUtf8Coder.of()))
      .apply(transform)
-     .apply(Write.to(sink));
+     .apply(write);
 
     p.run();
     assertThat(sinkContents, containsInAnyOrder(inputs.toArray()));
```

Comment:
Added a test that writes are balanced by making 1000 elts, 10 shards, and asserting that min and max are close. With printlns added, on the directrunner min was 99 and max was 101..so seems pretty good.